server/src/main/java/org/apache/druid/segment/realtime/appenderator/AppenderatorConfig.java

```diff
@@ -34,6 +34,13 @@
 
   boolean isSkipBytesInMemoryOverheadCheck();
 
+  /**
+   * Whether maximum memory usage should be considered in estimation.
+   */
+  default boolean isUseMaxMemoryEstimates() {
```

Comment:
Should this really be a default method instead of becoming a real thing on the config object?